partiql-tests-data/eval/primitives/operators/overlaps.ion

```diff
@@ -0,0 +1,470 @@
+      ],
+      output:true
+    }
+  },
+  {
+    name:"overlaps with date period containing another - returns true",
+    statement:"[DATE '2023-01-01', DATE '2023-01-31'] OVERLAPS [DATE '2023-01-10', DATE '2023-01-20']",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      output:true
+    }
+  },
+  {
+    name:"overlaps with single point date periods that match",
+    statement:"[DATE '2023-01-15', DATE '2023-01-15'] OVERLAPS [DATE '2023-01-15', DATE '2023-01-15']",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      output:true
+    }
+  },
+  {
+    name:"overlaps with single point date periods that don't match",
+    statement:"[DATE '2023-01-15', DATE '2023-01-15'] OVERLAPS [DATE '2023-01-16', DATE '2023-01-16']",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      output:false
+    }
+  },
+  {
+    name:"overlaps with overlapping date-interval period",
+    statement:"[DATE '2023-01-01', INTERVAL '30' DAY] OVERLAPS [DATE '2023-01-15', DATE '2023-02-15']",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      output:true
+    }
+  },
+  {
+    name:"overlaps with non-overlapping date-interval period",
+    statement:"[DATE '2023-01-01', INTERVAL '10' DAY] OVERLAPS [DATE '2023-01-15', DATE '2023-01-31']",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      output:false
+    }
+  },
+  {
+    name:"overlaps with touching date-interval period",
+    statement:"[DATE '2023-01-01', INTERVAL '14' DAY] OVERLAPS [DATE '2023-01-15', DATE '2023-01-31']",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      output:false
+    }
+  },
+  {
+    name:"overlaps with both periods using intervals",
+    statement:"[DATE '2023-01-01', INTERVAL '30' DAY] OVERLAPS [DATE '2023-01-15', INTERVAL '30' DAY]",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      output:true
+    }
+  },
+  {
+    name:"overlaps with negative interval",
+    statement:"[DATE '2023-01-31', INTERVAL '-30' DAY] OVERLAPS [DATE '2023-01-15', DATE '2023-02-15']",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      output:true
+    }
+  },
+  {
+    name:"overlaps with wrong collection size (first period has 1 element)",
+    statement:"[DATE '2023-01-01'] OVERLAPS [DATE '2023-01-01', DATE '2023-01-31']",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      output:null
+    }
+  },
```

Comment:
I think this should be an error in `EvalModeError` / PLK's `strict` mode and return missing in `EvalModeCoerce` / PLK's `permissive`. Similarly for the following two tests with mistyped arugments.